components/atom/button/src/Button.js

```diff
@@ -20,7 +21,14 @@ const Button = ({
   const rel = attrs.rel || defaultRel
 
   return link ? (
-    <Link {...attrs} href={href} target={target} rel={rel} ref={forwardingRef}>
+    <Link
+      {...attrs}
+      href={href}
+      target={target}
+      rel={rel}
+      ref={forwardingRef}
+      state={routeState}
```

Comment:
🤔  can't this be accomplished already via `attrs`? I'm not sure adding a non-standard `html` attribute here (`state`) makes sense when `linkFactory` is providing a regular anchor element... what do you think?

Reply:
The routing state management it's not a responsibility of button component.

Reply:
you should better use linkFactory prop abstraction